pandas/core/frame.py

```diff
@@ -716,24 +716,24 @@ def _repr_html_(self):
     def to_string(
         self,
         buf=None,
-        columns=None,
-        col_space=None,
-        header=True,
-        index=True,
-        na_rep="NaN",
-        formatters=None,
-        float_format=None,
-        sparsify=None,
-        index_names=True,
-        justify=None,
-        max_rows=None,
-        min_rows=None,
-        max_cols=None,
-        show_dimensions=False,
-        decimal=".",
-        line_width=None,
-        max_colwidth=None,
-    ):
+        columns: Optional[List[str]] = None,
+        col_space: Optional[Union[str, int]] = None,
+        header: Union[bool, List[str]] = True,
+        index: bool = True,
+        na_rep: str = "NaN",
+        formatters: Optional[fmt.formatters_type] = None,
```

Comment:
I think i made a mistake with the alias naming of `formatters_type` using python variable convention. we should agree a convention for type aliases. Typeshed tends to use Pascal case with a leading underscore since the aliases are not intended to be public.

also if we are reusing a alias, it could be moved to pandas._typing. in this case, the leading underscore should probably not be included as the alias is intended to be imported by other modules.

the dilemma here, is that following these strict conventions, results in a mixture of type aliases with underscores and without.  

@WillAyd ?

Reply:
In typescript it's common practice to indicate type interface with is leading `I`, sometimes a leading `_I`. Interface (similar to type alias in python) for some variable `Foo` will be named `IFoo` or `_IFoo`. It would be nice to have some similar convention here as well.

Reply:
Not sure I've come across anything official but most everything in the stdlib typing module is camel case, so I think can follow that format.

I'm OK with this being a follow up PR, though welcome to tackle here as well if you like

Reply:
Somewhat of a sidebar but I do find it confusing for the name to be `formatters_type` - would that be the same as `Type[formatters]`? If so I think should leverage the latter, and if not I think it might be confusing to append `_type` to things

Reply:
that's the other mistake i made in choosing a name!

probably `formatters` argument should have type `Formatters` if using an alias

and `float_format` argument would have type `FloatFormat`

fairly simple rule as long as no conflicts with classses.

Reply:
Will open a new PR sometime later. I will open a issue now for reminder, will check any other similar names too.

Reply:
#28480